docs/protocol/on-chain/security.mdx

```diff
@@ -0,0 +1,286 @@
+5. **Log state transitions**: Maintain audit trail of all <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip> updates
+
+**Sample Validation Checklist**:
+
+```markdown
+Before submitting state on-chain:
+☐ Verify all required signatures present
+☐ Verify signatures are valid for expected participants
+☐ Verify state version is sequential
+☐ Verify allocations sum correctly
+☐ Verify magic numbers (CHANOPEN/CHANCLOSE) if applicable
+☐ Verify channelId matches expected value
+☐ Test with small amounts first
+```
+
+### For Adjudicator Developers
+
+**Critical Requirements**:
+
+1. **Implement strict version comparison**: Ensure newer states always supersede older ones
+2. **Validate state transitions**: Enforce application-specific rules correctly
+3. **Optimize for gas efficiency**: Validation happens on-chain during disputes
+4. **Consider edge cases**: Handle all possible state transition scenarios
+5. **Audit thoroughly**: Security review before deployment is essential
+
+:::warning Adjudicator Responsibility
+<Tooltip content="A smart contract that validates state transitions according to application-specific rules.">Adjudicators</Tooltip> are critical to <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> security. A flawed adjudicator can undermine the entire channel's safety guarantees.
+:::
+
+**Adjudicator Development Checklist**:
+
+```markdown
+☐ Version comparison logic is correct and strict
+☐ All state transition rules are validated
+☐ Gas usage is optimized
+☐ Edge cases are handled (empty allocations, zero amounts, etc.)
+☐ Fallback behavior is safe (reject on ambiguity)
+☐ Code is audited by security professionals
+☐ Test coverage includes adversarial scenarios
+```
+
+## Security Guarantees Summary
+
+| Property | Guarantee | Mechanism |
+|----------|-----------|-----------|
+| **Funds Safety** | Cannot lose funds with valid signed state | Challenge-response + signatures |
+| **State Validity** | Only properly signed states accepted | ECDSA signature verification |
+| **Liveness** | Can always close if blockchain is live | Unilateral challenge mechanism |
+| **Censorship Resistance** | Any party can enforce closure | Multiple submission paths |
+| **No Replay** | Old states cannot be reused | Version number validation |
+
+:::success Strong Security Model
+<Tooltip content="The on-chain smart contract protocol.">Nitrolite</Tooltip> provides **strong security guarantees** equivalent to Layer 1 <Tooltip content="The underlying distributed ledger technology (e.g., Ethereum).">blockchain</Tooltip> security, while enabling Layer 2 scalability and efficiency.
```

Comment:
I would not specify that our security is equivalent, but rather "built on top of"

```suggestion
<Tooltip content="The on-chain smart contract protocol.">Nitrolite</Tooltip> provides **strong security guarantees** built on top of Layer 1 <Tooltip content="The underlying distributed ledger technology (e.g., Ethereum).">blockchain</Tooltip> security, while enabling Layer 2 scalability and efficiency.
```